travis/setup_dependencies_common.sh

```diff
@@ -313,7 +321,55 @@ if [[ ! -z $CONDA_DEPENDENCIES ]]; then
     fi
 fi
 
+if [[ ! -z $CONDA_DEPENDENCIES ]]; then
+    # Do a dry run of the conda install here to make sure that pins are
+    # ACTUALLY being respected. This will become unnecessary when
+    # https://github.com/conda/conda/issues/9052
+    # is fixed
+    dry_run_bad=`conda install --dry-run $CONDA_DEPENDENCIES 2>&1 | grep -c "conflicts with explicit specs"`
+    echo "dry_run_bad is $dry_run_bad"
+    if [[ $dry_run_bad -gt 0 ]]; then
+        echo "restoring free channel"
+        # Add the free channel, which might fix this...
+        conda config --set restore_free_channel true
+        # ...but only if the free channel is not ruled out by strict
+        # channel priority
+        if [[ ! -z $CONDA_CHANNEL_PRIORITY && $CONDA_CHANNEL_PRIORITY == strict ]]; then
+            # If the channel priority is strict we should fail instead of silently
+            # changing how the solve is done.
+            echo "cannot solve this environment with pinnings and strict channel priority"
+            exit 1
+        fi
+
+        # Try the dry run again, fail if pinnings are still ignored
+        dry_run_bad=$(conda install --dry-run $CONDA_DEPENDENCIES 2>&1 | grep -c "conflicts with explicit specs")
```

Comment:
you need the backticks here, too
```suggestion
        dry_run_bad=`conda install --dry-run $CONDA_DEPENDENCIES 2>&1 | grep -c "conflicts with explicit specs"`
```